source/support.tex

```diff
@@ -4538,183 +4484,410 @@
-as specified in ISO/IEC/IEEE 60559.
+  \tcode{t < u},
+  \tcode{t <= u},
+  \tcode{t > u},
+  \tcode{t >= u},
+  \tcode{u < t},
+  \tcode{u <= t},
+  \tcode{u > t}, and
+  \tcode{u >= t}
+  have the same domain.
+\item
+  \tcode{bool(t < u) == bool(u > t)} is \tcode{true},
 \item
-Otherwise, returns \tcode{a <=> b}
-if that expression is well-formed and
-convertible to \tcode{strong_ordering}.
+  \tcode{bool(u < t) == bool(t > u)} is \tcode{true},
 \item
-Otherwise, if the expression \tcode{a <=> b} is well-formed,
-then the function is defined as deleted.
+  \tcode{bool(t <= u) == bool(u >= t)} is \tcode{true}, and
 \item
-Otherwise, if the expressions \tcode{a == b} and \tcode{a < b}
-are each well-formed and convertible to \tcode{bool}, then
+  \tcode{bool(u <= t) == bool(t >= u)} is \tcode{true}.
+\end{itemize}
+
+\begin{codeblock}
+template <typename T, typename Cat = partial_ordering>
+  concept ThreeWayComparable =
+    @\placeholdernc{weakly-equality-comparable-with<T, T>}@ &&
+    (!ConvertibleTo<Cat, partial_ordering> || @\placeholdernc{partially-ordered-with}@<T, T>) &&
+    requires(const remove_reference_t<T>& a, const remove_reference_t<T>& b) {
+      { a <=> b } -> @\placeholdernc{compares-as}@<Cat>;
+    };
+\end{codeblock}
+
+\pnum
+Let \tcode{a} and \tcode{b} be lvalues
+of type \tcode{const remove_reference_t<T>}.
+\tcode{T} and \tcode{Cat}
+model \tcode{\libconcept{ThreeWayComparable}<T, Cat>} only if:
 \begin{itemize}
 \item
-if \tcode{a == b} is \tcode{true},
-returns \tcode{strong_ordering::equal};
+  \tcode{(a <=> b == 0) == bool(a == b)} is \tcode{true};
 \item
-otherwise, if \tcode{a < b} is \tcode{true},
-returns \tcode{strong_ordering::less};
+  \tcode{(a <=> b != 0) == bool(a != b)} is \tcode{true};
 \item
-otherwise,
-returns \tcode{strong_ordering::greater}.
-\end{itemize}
+  \tcode{((a <=> b) <=> 0)} and \tcode{(0 <=> (b <=> a))} are equal;
 \item
-Otherwise, the function is defined as deleted.
+  if \tcode{Cat} is convertible to \tcode{strong_equality}, \tcode{T} models
+  \libconcept{EqualityComparable}\iref{concept.equalitycomparable};
+\item
+  if \tcode{Cat} is convertible to \tcode{partial_ordering}:
+  \begin{itemize}
+  \item
+    \tcode{(a <=> b < 0) == bool(a < b)} is \tcode{true},
+  \item
+    \tcode{(a <=> b > 0) == bool(a > b)} is \tcode{true},
+  \item
+    \tcode{(a <=> b <= 0) == bool(a <= b)} is \tcode{true}, and
+  \item
+    \tcode{(a <=> b >= 0) == bool(a >= b)} is \tcode{true}; and
+  \end{itemize}
+\item
+  If \tcode{Cat} is convertible to \tcode{strong_ordering}, \tcode{T} models
+  \libconcept{StrictTotallyOrdered}\iref{concept.stricttotallyordered}.
 \end{itemize}
-\end{itemdescr}
 
-\indexlibrary{\idxcode{weak_order}}%
-\begin{itemdecl}
-template<class T> constexpr weak_ordering weak_order(const T& a, const T& b);
-\end{itemdecl}
-
-\begin{itemdescr}
 \pnum
-\effects
-Compares two values and produces a result of type \tcode{weak_ordering}:
-
```

Comment:
The definition of `ThreeWayComparableWith` should be right here, but is missing.

Reply:
Fixed.